morpho-checkpoint-node/src/Morpho/Node/Env.hs

```diff
@@ -113,14 +113,17 @@ initProducers :: NodeConfiguration -> IO [RemoteAddress]
 initProducers nc = do
   let file = unTopology $ ncTopologyFile nc
       nid = ncNodeId nc
-  Right (NetworkTopology topology) <- readTopologyFile file
-
-  case List.lookup nid $
-    map (\ns -> (CoreNodeId $ nodeId ns, producers ns)) topology of
-    Just ps -> return ps
-    Nothing -> do
-      hPutStrLn stderr $ "Node " <> Text.pack (show nid) <> " not found in topology file " <> Text.pack (show file)
+  mtopology <- readTopologyFile file
+  case mtopology of
+    Left err -> do
+      hPutStrLn stderr $ Text.pack err
```

Comment:
When there's an error in the JSON file, the error is:

```
Error in $: not enough input. Expecting ',' or ']'
```

What is that dollar sign?

Reply:
This apparently represents the root of a JSON path, see `Data.Aeson.Types.formatPath`. There doesn't appear to be an easy way to format it better

Reply:
Maybe say that the error is in the topology file?

Reply:
Oh yeah good point, I'll fix that

Reply:
Done that now